pkg/helm/controller.go

```diff
@@ -102,14 +109,21 @@ func (c *Controller) OnHelmChanged(key string, chart *helmv1.HelmChart) (*helmv1
 	job, valuesConfigMap, contentConfigMap := job(chart)
 	objs.Add(serviceAccount(chart))
 	objs.Add(roleBinding(chart))
-	objs.Add(job)
-	if valuesConfigMap != nil {
-		objs.Add(valuesConfigMap)
-	}
-	if contentConfigMap != nil {
-		objs.Add(contentConfigMap)
+
+	if config, err := c.confController.Cache().Get(chart.Namespace, chart.Name); err != nil {
+		if !errors.IsNotFound(err) {
+			logrus.Errorf("%s failed to get HelmChartConfig for %s: %s", Name, key, err)
```

Comment:
Is it expected that a chart will always be paired with a config? If not then this error message is extreme and would not be warranted even at the WARN log level. Otherwise, if a chart is expected to always have a config, but nothing breaks when a config is missing, maybe this should be a WARN instead.

Reply:
The error is only printed `if !errors.IsNotFound(err)` so a HelmChart without a HelmChartConfig will not cause anything to be printed. It would need to be some sort of more severe backend error.

Reply:
hah, i missed the negation. i blame a lack of caffeine

Reply:
Actually, shouldn't we return an error here? The implication is that it was found and we couldn't get it. To me, that's an error that we should bubble up, not just log and move on.

Reply:
No, the Get() returns a NotFound error if the resource doesn't exist. This is acceptable, because not every HelmChart will have a HelmChartConfig. If it IS NOT a NotFound error, we print it. If it IS a NotFound error, we ignore it because that's OK.

Reply:
I'm saying that your logic when it IS NOT a notFound error has a problem. Generally, we should return the error so that the framework can retry, not print it and continue on with the logic.

Reply:
Ah OK. So you'd prefer to return non-NotFound errors rather than just print and continue. Can do!